lib/debug/frame_info.rb

```diff
@@ -57,11 +57,13 @@ def frame_type
       end
     end
 
+    BLOCK_LABL_REGEXP = /block( \(\d+ levels\))* in (.+)/
+
     def block_identifier
       return unless frame_type == :block
       args = parameters_info(iseq.argc)
-      _, _, block_loc = location.label.split(" ")
-      [block_loc, args]
+      _, level, block_loc = location.label.match(BLOCK_LABL_REGEXP).to_a
+      [level || "", block_loc, args]
```

Comment:
you can get this information by comparing with `location.labal` and `location.base_label` without regexp (my original code used it).

Reply:
can we extract string this way in Ruby?

```ruby
s1 = "block (2 levels) in <main>"
s2 = "block in <main>"

# s2 - s1 == "(2 levels)" ?
```

I believe the original implementation was:

```ruby
label_prefix = frame.location.label.sub('block'){ "block#{args_str}" }
```

which works well. 

but since we colorize components differently, we need to extract them separately. 
I also need to reorder the level & block location info:

```ruby
original = "block (2 levels) in <main>"
updated = "block in <main> (2 levels)" 
```

to make sure block with arguments looks better

```
block {|ten=10|} in second_call (2 levels) at target.rb:9
# instead of 
block {|ten=10|} (2 levels) in second_call at target.rb:9 # this looks confusing 
```

Reply:
thank you it was my misremembering. 
ok. please fix regexp to use `\A` and `\z` to reduce misinterpretation. (we can name Ruby's method with any names).

Reply:
👍 fixed